### DISCUSSION TITLE
Check for updates to pinned plugins

### DISCUSSION THREAD

**Discussion by jameshiew** [wbthomason/packer.nvim#586] [Ideas]
If you pin a plugin to a certain release (using something like `tag = 'v0.2'`), you may want to get notified when a new release becomes available. I'm not sure of the best workflow for this, maybe a prompt when `:PackerSync` or `:PackerUpdate` runs? Ideally you could auto-update to the new release as well.

**Comment by wbthomason** [comment]
Right now, `packer` does not do anything to check for new tags, but iirc this was mentioned in another feature request. We could add an option to check if there are tags pinned to commits after the current tag in the update logic fairly easily. The UX for prompting to update is a bit trickier - if someone has pinned to e.g. `tag = '*'` for the latest tag, then we can auto-update, but otherwise we may just want to display a notification (which would need to be configurable so that users aren't spammed for plugins they don't want to upgrade).

**Reply by jameshiew** [reply]
UX wise, an indication of package versions in the `:PackerStatus` view could be helpful, maybe with a hint like a `v0.2 -> v0.3` next to any pinned packages which have an update available. Then a keypress could effect an update.  It would still be a bit manual and rely on users using `:PackerStatus` every so often, though it could be coupled with  echoing a brief message on running `:PackerUpdate`/`:PackerSync` if any pinned packages do have updates.  I think spacemacs has something similar iirc (it emits a message like '52 packages have updates').

**Reply by wbthomason** [reply]
Yeah, spacemacs has it a bit easier because all of the packages it reports update status for come from MELPA/ELPA/similar (per my understanding), which I believe has an API for getting package versions, etc.

We could add an optional feature to check for updates upon running `PackerStatus`, or check for pinned plugins and display a message, as you suggest. If you'd like to take a stab at this, I'd be happy to guide a PR (though I'll warn that the `git` code in `packer` is probably the most in need of a rewrite).
